exec/vector/src/main/java/org/apache/drill/exec/vector/accessor/writer/ScalarArrayWriter.java

```diff
@@ -60,13 +60,17 @@
     public final void nextElement() { next(); }
   }
 
-  private final BaseScalarWriter elementWriter;
+  private final ConcreteWriter elementWriter;
 
   public ScalarArrayWriter(ColumnMetadata schema,
       RepeatedValueVector vector, BaseScalarWriter elementWriter) {
     super(schema, vector.getOffsetVector(),
         new ScalarObjectWriter(elementWriter));
-    this.elementWriter = elementWriter;
+
+    // Save the writer from the scalar object writer created above
+    // which may have wrapped the element writer in a type convertor.
+
+    this.elementWriter = (ConcreteWriter) elementObjWriter.scalar();
```

Comment:
Yes. The plumbing ensures that the actual implementation is always of this class. This is, however, part of the messiness described in a comment elsewhere. I'm think about how to refactor this to avoid the need for these casts, for the clunky pass-through methods in the type converter, and to make the change fit into the result set loader. This is a first step.